### ISSUE TITLE
PreparedBatch::execute closes context + deregisters argument factories

### ISSUE THREAD

**Issue by skradel** [jdbi/jdbi#1651]
When more than one PreparedBatch is constructed and executed for a single Handle, the second PreparedBatch is unable to bind arguments that rely on plugins; for example, Jdbi3 configured with the PostgresPlugin is unable to bind an Array<String> after the first PreparedBatch executes.

Consider the following Scala example that chunks an iterator of POJOs into PreparedBatches.  bindModel is a simple function that manually binds object properties to named query parameters, the type of one of which is Array<String>.  The first batch is OK; the second batch dies with `No argument factory registered for '[Ljava.lang.String;`

`jdbi.useTransaction(handle => {
      client.iterator[T]()
        .grouped(500)
        .foreach(xs => {
          xs.foldLeft(handle.prepareBatch(queryText))(
            bindModel(_, _).add())
            .execute()
        })
    })`

Jdbi 3.11.1 (via dropwizard 2.0.0)

**Comment by stevenschlansker** [comment]
Hi @skradel , thanks for reporting this issue.
I wrote a small test case:
```java
    @Test
    public void testDoubleBatchInsert() {
        h.useTransaction(tx -> {
            final Runnable insertBatch =
                () -> tx.prepareBatch("INSERT INTO foo (bar) SELECT unnest(:vals)")
                    .bind("vals", new UUID[] { UUID.randomUUID(), UUID.randomUUID() })
                    .execute();
            insertBatch.run();
            insertBatch.run();
            assertThat(h.createQuery("SELECT * FROM foo")
                    .mapTo(UUID.class)
                    .list())
            .hasSize(4);
        });
    }
```
This test runs two batches back to back in the same transaction, and passes against jdbi master.

Could you provide a small self-contained example showing the problem, preferably in Java?  It may not be as simple as just closing a context.  Thanks!

**Comment by skradel** [comment]
@stevenschlansker Sure thing, it will be a little work to put everything together exactly - of note the bound type is String[] - I don't know if UUID[] might behave differently.

**Comment by hgschmie** [comment]
We shipped a lot of changes with the 3.35 and beyond releases. As the test from @stevenschlansker works, I assume that this has gone stale and can be closed. Please reopen if that problem still exists with current versions of Jdbi.
